server/src/main/java/com/cloud/server/StatsCollector.java

```diff
@@ -718,7 +718,7 @@ protected void runInContext() {
                  getDynamicDataFromDB();
                  long interval = (Long) dbStats.get(uptime) - lastUptime;
                  long activity = (Long) dbStats.get(queries) - lastQueries;
-                 loadHistory.add(0, Double.valueOf(activity / interval));
+                 loadHistory.add(0, interval == 0 ? 0 : Double.valueOf(activity / interval));
```

Comment:
clgtm, as in this will avoid the NPE, but should we add avalue of 0 at all in this case? It seems to me that uptime - lastUptime == 0, we have another problem and should not add a value.

Reply:
@lucas-a-martins (cc @Pearl1594 @rohityadavcloud ) please reconsider.

Reply:
if activities == 5 we don't want the user to think 0 because af a possible math error.
```suggestion
                 loadHistory.add(0, interval == 0 ? -1 : Double.valueOf(activity / interval));
```